File: packages/api/aqapi/language/domain/value_object/language_name.py
from dataclasses import dataclass
from typing import override

from aqapi.core.domain.value_object.base_value_object import BaseValueObject
from aqapi.core.messages.locale_string import LocaleString


@dataclass
class LanguageName(BaseValueObject):
    """言語名の値オブジェクト"""
    
    _value: str

    def __init__(self, value: str):
        super().__init__(value)

    def _validate(self) -> None:
        self._validator.required()

    @property
    @override
    def _value_name(self) -> LocaleString:
        return LocaleString(
            ja="言語名",
            en="Language Name"
        )

File: packages/flutter/lib/language/enum/value_object/language_name.dart
import 'package:allowance_questboard/core/messages/locale_string.dart';
import 'package:allowance_questboard/core/value_object/base_state_object.dart' show BaseValueObject;

/// 言語名の値オブジェクト
class LanguageName extends BaseValueObject<String> {
  LanguageName(super.value);
  
  @override
  void validate() {
    validator.required();
    validator.minLength(1);
    validator.maxLength(100);
  }
  
  @override
  LocaleString get valueName => const LocaleString(
    ja: "言語名",
    en: "Language Name",
  );
}
